Windows MSVC C++17 CMake/Ninja Compile Errors on prebuilt header

Using prebuilt https://aurora-opensource.github.io/au/0.3.2/au_noio.hh, I included the file in my project and I get this block of errors. 
```
C:\Users\conno\Documents\SC\Platform\Compute_Server\externals\Au\au.h(1247): error C2672: 'au::detail::are_consecutive_elements_in_order_for': no matching overloaded function found
  C:\Users\conno\Documents\SC\Platform\Compute_Server\externals\Au/au.h(1247): note: see reference to class template instantiation 'au::AreElementsInOrder<Pack,Pack<Ts...>>' being compiled
C:\Users\conno\Documents\SC\Platform\Compute_Server\externals\Au\au.h(1247): error C3207: 'au::detail::are_consecutive_elements_in_order_for': invalid template argument for 'Pack', class template expected
  C:\Users\conno\Documents\SC\Platform\Compute_Server\externals\Au/au.h(1229): note: see declaration of 'au::detail::are_consecutive_elements_in_order_for'
C:\Users\conno\Documents\SC\Platform\Compute_Server\externals\Au\au.h(1247): error C2977: 'au::detail::are_consecutive_elements_in_order_for': too many template arguments
  C:\Users\conno\Documents\SC\Platform\Compute_Server\externals\Au/au.h(1219): note: see declaration of 'au::detail::are_consecutive_elements_in_order_for'
C:\Users\conno\Documents\SC\Platform\Compute_Server\externals\Au\au.h(1261): error C2440: '<function-style-cast>': cannot convert from 'initializer list' to 'std::array<bool,sizeof...(Ts)>'
  C:\Users\conno\Documents\SC\Platform\Compute_Server\externals\Au/au.h(1261): note: The target type has no constructors
  C:\Users\conno\Documents\SC\Platform\Compute_Server\externals\Au/au.h(1261): note: see reference to class template instantiation 'au::AreAllPowersNonzero<Pack,Pack<Ts...>>' being compiled
```

I am unable to figure out what the cause might be. It seems this may be the only problem with Au on this compiler, though, so it's valuable to know a fix or workaround.

### Compile info:
- Windows 10
- RelWithDebInfo Configuration
- Toolchain: msvc_x64_64
- MSVC v142 - VS 2019
- cl version 14.29.30133
- CMake build through Visual Studio's CMake support (Ninja)
- CMake configuration:
```cmake
set(CMAKE_CXX_STANDARD 17) 
set(CMAKE_CXX_STANDARD_REQUIRED ON)
set(CMAKE_CXX_EXTENSIONS OFF)
set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```

--- a/au/math.hh
+++ b/au/math.hh
@@ -17,6 +17,7 @@
 #include <algorithm>
 #include <cmath>
 #include <limits>
+#include <type_traits>
 
 #include "au/quantity.hh"
 #include "au/quantity_point.hh"

--- a/au/math.hh
+++ b/au/math.hh
@@ -14,6 +14,7 @@
 
 #pragma once
 
+#include <algorithm>
 #include <cmath>
 #include <limits>
 

--- a/au/packs.hh
+++ b/au/packs.hh
@@ -14,7 +14,6 @@
 
 #pragma once
 
-#include <array>
 #include <cstdint>
 #include <ratio>
 #include <utility>
@@ -519,38 +518,34 @@ struct IsValidPack : stdx::conjunction<detail::IsPackOf<Pack, T>,
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 // `AreElementsInOrder` implementation.
 
-namespace detail {
-
 template <template <class...> class Pack>
-constexpr bool are_consecutive_elements_in_order_for() {
-    return true;
-}
+struct AreElementsInOrder<Pack, Pack<>> : std::true_type {};
 
 template <template <class...> class Pack, typename T>
-constexpr bool are_consecutive_elements_in_order_for() {
-    return true;
-}
+struct AreElementsInOrder<Pack, Pack<T>> : std::true_type {};
 
 template <template <class...> class Pack, typename T1, typename T2, typename... Ts>
-constexpr auto are_consecutive_elements_in_order_for() {
-    return InOrderFor<Pack, T1, T2>::value &&
-           are_consecutive_elements_in_order_for<Pack, T2, Ts...>();
-}
+struct AreElementsInOrder<Pack, Pack<T1, T2, Ts...>>
+    : stdx::conjunction<InOrderFor<Pack, T1, T2>, AreElementsInOrder<Pack, Pack<T2, Ts...>>> {};
 
-template <std::size_t N>
-constexpr bool all_true(const std::array<bool, N> &values) {
-    for (auto i = 0u; i < values.size(); ++i) {
-        if (!values[i]) {
+namespace detail {
+
+constexpr bool all_true() { return true; }
+
+template <typename... Predicates>
+constexpr bool all_true(Predicates &&...values) {
+    // The reason we bother to make an array is so that we can iterate over it.
+    const bool value_array[] = {values...};
+
+    for (auto i = 0u; i < sizeof...(Predicates); ++i) {
+        if (!value_array[i]) {
             return false;
         }
     }
+
     return true;
 }
 }  // namespace detail
-
-template <template <class...> class Pack, typename... Ts>
-struct AreElementsInOrder<Pack, Pack<Ts...>>
-    : stdx::bool_constant<detail::are_consecutive_elements_in_order_for<Pack, Ts...>()> {};
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 // `AreBasesInOrder` implementation.
@@ -563,8 +558,7 @@ struct AreBasesInOrder<Pack, Pack<Ts...>> : AreElementsInOrder<Pack, Pack<BaseT<
 
 template <template <class...> class Pack, typename... Ts>
 struct AreAllPowersNonzero<Pack, Pack<Ts...>>
-    : stdx::bool_constant<detail::all_true(
-          std::array<bool, sizeof...(Ts)>{(ExpT<Ts>::num != 0)...})> {};
+    : stdx::bool_constant<detail::all_true((ExpT<Ts>::num != 0)...)> {};
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 // `SimplifyBasePowersT` implementation.

